validation-test/stdlib/FloatingPointConversion.swift.gyb

```diff
@@ -0,0 +1,192 @@
+  let input = ${OtherFloat}.greatestFiniteMagnitude 
+  var result = ${Self}(exactly: input)
+  expectNotEmpty(result)
+  var resultConvertedBack = ${OtherFloat}(result!)
+  expectEqual(input, resultConvertedBack)
+}
+
+/// Never-nil conversion from ${OtherFloat}.nextUp to ${Self}.
+FloatingPointConversionFailures.test("${OtherFloat}To${Self}Conversion/dest=${OtherFloat}.1.0.nextUp") {
+  // Test that nothing interesting happens and we end up with a non-nil, identical result.
+  let input = (1.0 as ${OtherFloat}).nextUp
+  var result = ${Self}(exactly: input)
+  expectNotEmpty(result)
+  var resultConvertedBack = ${OtherFloat}(result!)
+  expectEqual(input, resultConvertedBack)
+}
+
+%   else:
+
+/// Always-truncating conversion from ${OtherFloat}.nextUp to ${Self}.
+FloatingPointConversionTruncations.test("${OtherFloat}To${Self}Conversion/dest=${OtherFloat}.greatestFiniteMagnitude") {
+  // Test that we check if we truncate floats not representable by another type.
+  let input = ${OtherFloat}.greatestFiniteMagnitude 
+  var result = ${Self}(input)
+  var resultConvertedBack = ${OtherFloat}(result)
+  expectNotEqual(input, resultConvertedBack)
+}
+
+/// Always-succeeding, but out-of-range conversion from ${OtherFloat}.greatestFiniteMagnitude to ${Self}.
+FloatingPointConversionTruncations.test("${OtherFloat}To${Self}Conversion/dest=${OtherFloat}.1.0.nextUp") {
+  // Test that we check if we truncate floats not representable by another type.
+  let input = (1.0 as ${OtherFloat}).nextUp
+  var result = ${Self}(input)
+  var resultConvertedBack = ${OtherFloat}(result)
+  expectNotEqual(input, resultConvertedBack)
+}
+
+/// Always-nil failable conversion from ${OtherFloat}.greatestFiniteMagnitude to ${Self}.
+FloatingPointConversionFailures.test("${OtherFloat}To${Self}FailableConversion/dest=${OtherFloat}.greatestFiniteMagnitude") {
+  // Test that we check if we return nil when a float would be truncated in conversion.
+  let input = ${OtherFloat}.greatestFiniteMagnitude 
+  var result = ${Self}(exactly: input)
+  expectEmpty(result)
+}
+
+/// Always-nil failable conversion from ${OtherFloat}.nextUp to ${Self}.
+FloatingPointConversionFailures.test("${OtherFloat}To${Self}FailableConversion/dest=${OtherFloat}.1.0.nextUp") {
+  // Test that we check if we return nil when a float would be truncated in conversion.
+  let input = (1.0 as ${OtherFloat}).nextUp
+  var result = ${Self}(exactly: input)
+  expectEmpty(result)
+}
+
+%   end
+
+FloatingPointConversionTruncations.test("${OtherFloat}To${Self}Conversion/dest=NaN") {
+  // Check that initializing with NaN from another type works.
+  let input = ${OtherFloat}.nan
+  var result = ${Self}(input)
+  var resultConvertedBack = ${OtherFloat}(result)
+  expectTrue(input.isNaN)
+  expectTrue(resultConvertedBack.isNaN)
+}
+
+FloatingPointConversionFailures.test("${OtherFloat}To${Self}Conversion/dest=NaN") {
+  // Check that initializing with NaN doesn't fail the initializer.
```

Comment:
The comment is now stale.  I feel like you can delete it, the test code speaks for itself.